### PR TITLE
Fix/rose extension mark

### PR DESCRIPTION
**Pull request by skie1997**
[[中文版模板 / Chinese template](https://github.com/VisActor/VChart/blob/main/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Refactoring
- [ ] Update dependency
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Release
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Workflow
- [ ] Other (about what?)

### 🐞 Bugserver case id
6572e0cb9464710a7ba9dc8f

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |           |
| 🇨🇳 Chinese |           |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [ ] Doc is updated/provided or not needed
- [ ] Demo is updated/provided or not needed
- [ ] TypeScript definition is updated/provided or not needed
- [ ] Changelog is provided or not needed

---

### 🚀 Summary

copilot:summary

### 🔍 Walkthrough

copilot:walkthrough

### CODE REVIEW

File: packages/vchart/src/series/rose/rose.ts [hunk 1]
```diff
@@ -39,6 +39,20 @@ export class RoseSeries<T extends IRoseSeriesSpec = IRoseSeriesSpec> extends Ros
     this.initRoseMarkStyle();
   }
 
+  protected _buildMarkAttributeContext() {
+    super._buildMarkAttributeContext();
+    // center
+    this._markAttributeContext.getCenter = () => {
```
Comment (kkxxkk2019) [inline packages/vchart/src/series/rose/rose.ts:45]: 这个应该对于 polar series 都适用把

File: packages/vchart/src/series/rose/rose.ts [hunk 2]
```diff
@@ -39,6 +39,20 @@ export class RoseSeries<T extends IRoseSeriesSpec = IRoseSeriesSpec> extends Ros
...
+      return {
+        x: () => this.angleAxisHelper.center().x,
+        y: () => this.angleAxisHelper.center().y
+      };
+    };
+    // angle scale
+    this._markAttributeContext.startAngleScale = datum => this.startAngleScale(datum);
```
Comment (kkxxkk2019) [inline packages/vchart/src/series/rose/rose.ts:52]: 饼图是不是也适用？ 增加参数的时候需要系统性考虑下

File: packages/vchart/src/series/rose/rose.ts (hunk 1 above)
Comment (skie1997) [reply packages/vchart/src/series/rose/rose.ts:45]: 饼图有自己的center逻辑，它在做映射的时候没有依赖angleAxisHelper，所以没有写到polar series作为统一逻辑
image
image

File: packages/vchart/src/series/rose/rose.ts (hunk 2 above)
Comment (skie1997) [reply packages/vchart/src/series/rose/rose.ts:52]: 饼图的angle计算是在vgrammar层算的，同样没有依赖axisHelper，并且直接从datum里面就可以取到angle值，所以在这里没有给饼图导出这个参数。
image

不过为了统一性应该还是需要加下这个参数，到时候就不需要从datum里面拿angle了，直接用angleScale(datum）会比较好

Comment (skie1997) [reply packages/vchart/src/series/rose/rose.ts:52]: 这个我加一下

**Review by kkxxkk2019 — APPROVED** [review]

**Pull request merged**